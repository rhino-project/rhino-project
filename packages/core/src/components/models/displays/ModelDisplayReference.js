import { useCallback } from 'react';
import DisplayString from 'rhino/components/forms/displays/DisplayString';
import { useGlobalOverrides } from 'rhino/hooks/overrides';

const ModelDisplayReferenceBase = ({ model, ...props }) => {
  const accessor = useCallback((value) => value?.display_name, []);

  return <DisplayString accessor={accessor} {...props} />;
};

const defaultComponents = { ModelDisplayReference: ModelDisplayReferenceBase };

const ModelDisplayReference = ({ overrides, ...props }) => {
  const { ModelDisplayReference } = useGlobalOverrides(
    defaultComponents,
    overrides,
    props
  );

  return <ModelDisplayReference {...props} />;
};

export default ModelDisplayReference;
