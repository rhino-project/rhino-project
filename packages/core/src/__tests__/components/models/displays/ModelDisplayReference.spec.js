import ModelDisplayReference from 'rhino/components/models/displays/ModelDisplayReference';
import { sharedDisplayTests } from './sharedDisplayTests';

describe('ModelDisplayReference', () => {
  sharedDisplayTests(ModelDisplayReference);
});
