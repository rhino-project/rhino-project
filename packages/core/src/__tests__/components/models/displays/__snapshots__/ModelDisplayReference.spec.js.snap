// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelDisplayReference should render the empty text when value is nullish 1`] = `
<DocumentFragment>
  <input
    aria-invalid="false"
    autocomplete="off"
    class="form-control"
    id="dummy"
    name="dummy"
    path="dummy"
    readonly=""
    type="text"
    value="-"
  />
</DocumentFragment>
`;

exports[`ModelDisplayReference should render the overridden empty text when value is nullish 1`] = `
<DocumentFragment>
  <input
    aria-invalid="false"
    autocomplete="off"
    class="form-control"
    id="dummy"
    name="dummy"
    path="dummy"
    readonly=""
    type="text"
    value="baz"
  />
</DocumentFragment>
`;

exports[`ModelDisplayReference should render with global override shorthand 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;

exports[`ModelDisplayReference should render with global override shorthand for model 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;

exports[`ModelDisplayReference should render with global override shorthand for model and attribute 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;
